src/flows/shared/steps/OrganizeCollection/Editor/StagingArea.tsx

```diff
@@ -72,63 +251,145 @@ function StagingArea({ tokensRef, stagedItems }: Props) {
-                    size={size}
-                    mini={columns > 4}
-                  />
-                );
-              }
-              return (
-                <SortableStagedWhitespace key={stagedItem.id} id={stagedItem.id} size={size} />
-              );
-            })}
-          </StyledStagedNftContainer>
+        {/* Handles sorting for sections */}
+        <SortableContext items={sectionIds} strategy={verticalListSortingStrategy}>
+          {sectionIds.map((sectionId) => (
+            <DroppableSection
+              key={sectionId}
+              id={sectionId}
+              items={localStagedCollection[sectionId].items}
+              columns={localStagedCollection[sectionId].columns}
+            >
+              {/* Handles sorting for items in each section */}
+              <SortableContext
+                items={localStagedCollection[sectionId].items.map((item) => item.id)}
+              >
+                {localStagedCollection[sectionId].items.map((item) => {
+                  const columns = localStagedCollection[sectionId].columns;
+                  const size = IMAGE_SIZES[columns];
+                  const stagedItemRef = nftFragmentsKeyedByID[item.id];
+                  if (isEditModeToken(item) && stagedItemRef) {
+                    return (
+                      <SortableStagedNft
+                        key={item.id}
+                        tokenRef={stagedItemRef}
+                        size={size}
+                        mini={columns > 4}
+                      />
+                    );
+                  }
+                  return <SortableStagedWhitespace key={item.id} id={item.id} size={size} />;
+                })}
+              </SortableContext>
+            </DroppableSection>
+          ))}
         </SortableContext>
-        <DragOverlay adjustScale dropAnimation={defaultDropAnimationConfig}>
-          {activeItem && activeItemRef ? (
-            <StagedItemDragging
-              tokenRef={activeItemRef}
-              isEditModeToken={isEditModeToken(activeItem)}
-              size={itemWidth}
-            />
+        <DragOverlay dropAnimation={dropAnimation}>
+          {activeId ? (
+            sectionIds.includes(activeId as string) ? (
+              <SectionDragging
+                items={localStagedCollection[activeId].items}
+                itemWidth={IMAGE_SIZES[localStagedCollection[activeId].columns]}
+                columns={localStagedCollection[activeId].columns}
+                nftFragmentsKeyedByID={nftFragmentsKeyedByID}
+              />
+            ) : (
+              activeItem && (
+                <StagedItemDragging
+                  tokenRef={activeItemRef || null}
+                  isEditModeToken={isEditModeToken(activeItem)}
+                  size={
+                    IMAGE_SIZES[
+                      findSection(activeId)
+                        ? localStagedCollection[findSection(activeId)!].columns
```

Comment:
nit: if you want to avoid using `findSection(activeId)!`, in the `findSection` func you can throw an error instead of returning `undefined`. that way the type signature will say it always returns a string

Reply:
done!